### DISCUSSION TITLE
Using the formatter binary

### DISCUSSION THREAD

**Discussion by astrale-sharp** [nvarner/typst-lsp#417] [General]
We could either have good typstfmt config integration or simply let go of that, bundle a typstfmt binary or simply download it and let the user worry about the config.

Thoughts options?

**Comment by KillTheMule** [comment]
At least for (neo)vim I'd think you want support for documents that don't exist as a file, so that would complicate matters (not that you can't write it out, format it, send it back I suppose). Also, permissions might get hairy... I'd stay with using typstfmt as a library.

**Reply by astrale-sharp** [reply]
I mean typstfmt can take from stdin and give to stdout so that would probably make things pretty simple right?

**Reply by astrale-sharp** [reply]
What permissions are you referring to?

**Reply by KillTheMule** [reply]
> I mean typstfmt can take from stdin and give to stdout so that would probably make things pretty simple right?

Yeah right.

> What permissions are you referring to?

The permissions of the typstfmt binary. Downloading things automatically/bundling them means you have to deal with them. Maybe it's easier than I think :)

**Reply by astrale-sharp** [reply]
I think it's pretty common, lots of vscode extensions make me aware they are downloading stuff XD
bit shocking that binary run like that on my computer maybe :thinking:
